google/codelab/provision_spinnaker_codelab_image.sh

```diff
@@ -0,0 +1,83 @@
+#!/bin/bash
+
+# Copyright 2016 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script provisions a fresh spinnaker image with a Jenkins
+# server and an aptly repository, served on nginx. It also
+# provides a basic Jenkins configuration. This is to be used
+# specifically for creating a new (running) codelab image out of a fresh
+# spinnaker image.
+
+if [[ `/usr/bin/id -u` -ne 0 ]]; then
+  echo "$0 must be executed with root permissions; exiting"
+  exit 1
+fi
+
+/opt/spinnaker/bin/stop_spinnaker.sh
+service spinnaker stop
+service apache2 stop
+
+# update apt
+apt-get update
+apt-get -y upgrade
+
+# acquire and configure jenkins
+apt-get install -y git
+wget http://pkg.jenkins-ci.org/debian/binary/jenkins_2.1_all.deb
+dpkg -i jenkins_2.1_all.deb
+apt-get -f -y install
+sed -i "s/HTTP_PORT=.*/HTTP_PORT=9090/" /etc/default/jenkins
+
+# as jenkins, configure aptly
+cd /var/lib/jenkins
+wget https://dl.bintray.com/smira/aptly/0.9.5/debian-squeeze-x64/aptly
+chown jenkins /var/lib/jenkins/aptly
+sudo -u jenkins -H sh -c "chmod +x aptly"
+sudo -u jenkins -H sh -c "/var/lib/jenkins/aptly repo create hello"
+sudo -u jenkins -H sh -c '/var/lib/jenkins/aptly publish repo -architectures="amd64,i386" -component=main -distribution=trusty -skip-signing=true hello'
+
+# as jenkins, configure jenkins config directory
+# this storage bucket is public so we can pull the jenkins config from anywhere
+sudo -u jenkins -H sh -c 'wget https://storage.googleapis.com/codelab-jenkins-configuration/jenkins_dir.tar.gz'
+sudo -u jenkins -H sh -c 'tar -zxvf jenkins_dir.tar.gz'
+sudo -u jenkins -H sh -c 'git clone https://github.com/kenzanlabs/hello-karyon-rxnetty.git'
+
+# configure nginx to serve the aptly repo and start it
+apt-get install -y nginx=1.4.6-1ubuntu3.4
+cat > /etc/nginx/sites-enabled/default <<EOF
```

Comment:
We are assuming a fresh image, but I will add a line that backs up this file if it exists for some reason.